q2_diversity/_alpha/_visualizer.py

```diff
@@ -291,20 +287,26 @@ def _compute_rarefaction_data(feature_table, min_depth, max_depth, steps,
     data = {k: pd.DataFrame(np.NaN, index=rows, columns=cols)
             for k in metrics}
 
-    for d, i in itertools.product(depth_range, iter_range):
-        rt = rarefy(feature_table, d)
-        for m in metrics:
-            if m in phylogenetic_metrics():
-                # need a new rarefied table here in case a phylogenetic
-                # metric comes before a non-phylogenetic metric
-                rt_p = transform(rt, to_type=BIOMV210Format,
-                                 from_type=biom.Table)
-                vector = alpha_phylogenetic(table=rt_p, metric=m,
-                                            phylogeny=phylogeny)
-            else:
-                vector = alpha(table=rt, metric=m)
-            data[m][(d, i)] = vector
-    return data
+    if phylogeny:
+        phylogeny = qiime2.Artifact.import_data('Phylogeny[Rooted]', phylogeny)
```

Comment:
If we move this into the `scope` context manager, then you can use `make_artifact`, which tracks the lifetime of the artifact for you:

https://github.com/qiime2/qiime2/blob/55f122c689d647743c207d9891adcbf1def40897/qiime2/sdk/context.py#L41-L52

Reply:
resolved in 171f005